### PR TITLE
EVG-6465 valiating filtering, added more tests.

### PR DESCRIPTION
**Pull request by jimoleary**

### CODE REVIEW

**Comment by jimoleary** [comment]
Filtering was already implemented, I added some tests andfixed a constant.

File: model/reliability/query_test.go [hunk 1]
```diff
@@ -252,6 +291,144 @@ func TestGetTaskStatsTwoDocuments(t *testing.T) {
...
+			for testName, testCase := range map[string]func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter){
+				"No Matches": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						docs, err := GetTaskReliabilityScores(filter)
+						require.NoError(err)
+						require.NotEqual(len(docs), 0)
+					})
+				},
+				"Invalid Combination task1": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
```
Comment (lydiastepanek) [inline model/reliability/query_test.go]: Confused...Why is task1 + variant2 an invalid combination?

File: model/reliability/query_test.go [hunk 2]
```diff
@@ -252,6 +291,144 @@ func TestGetTaskStatsTwoDocuments(t *testing.T) {
...
+						require.NoError(err)
+						require.Len(docs, 2)
+						for _, doc := range docs {
+							require.Equal(doc.TaskName, task1)
+							require.NotEqual(doc.BuildVariant, variant2)
+							require.NotEqual(doc.Distro, distro2)
+						}
+					})
+				},
+				"task2": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
```
Comment (lydiastepanek) [inline model/reliability/query_test.go]: Isn't testing task1 then task2 testing the same thing? (that `GetTaskReliabilityScores` returns a score matching a filter containing Tasks, BuildVariants, and Distros)

File: model/reliability/query_test.go [hunk 3]
```diff
@@ -252,6 +291,144 @@ func TestGetTaskStatsTwoDocuments(t *testing.T) {
...
+					})
+				},
+				"MaxQueryLimit": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						task3 := "task3"
+						variantFmt := "variant %04d"
+						distroFmt := "distro %04d"
+
+						require.NoError(insertManyDailyTaskStats(MaxQueryLimit, project, "r1", task3, variantFmt, distroFmt, day1, 10, 5, 1, 1, 1, 2, 10.5))
```
Comment (lydiastepanek) [inline model/reliability/query_test.go]: Would it make sense to insert MaxQueryLimit + 1 stats to test that the query correctly returns 1000 docs and not 1001?

File: model/reliability/query_test.go (hunk 2 above)
Comment (jimoleary) [reply model/reliability/query_test.go]: Yes, I'm just being a bit paranoid that the task matching portion is working.

Comment (lydiastepanek) [reply model/reliability/query_test.go]: Ok

File: model/reliability/query_test.go [hunk 4]
```diff
@@ -67,6 +71,41 @@ func insertDailyTaskStats(project string, requester string, taskName string, var
...
+		return fmt.Sprintf(format, i)
+	}
+	return format
+}
+
+func insertManyDailyTaskStats(many int, projectFmt string, requesterFmt string, taskNameFmt string, variantFmt string, distroFmt string, date time.Time, numSuccess, numFailed, numTimeout, numTestFailed, numSystemFailed, numSetupFailed int, avgDuration float64) error {
+
+	items := []interface{}{}
+	for i := 0; i < many; i++ {
+		items = append(items, bson.M{
```
Comment (tychoish) [inline model/reliability/query_test.go]: is there a type other than a map for this?

File: model/reliability/query_test.go [hunk 5]
```diff
@@ -67,6 +71,41 @@ func insertDailyTaskStats(project string, requester string, taskName string, var
...
+	n := strings.Count(format, "%")
+	if n > 0 {
+		return fmt.Sprintf(format, i)
+	}
+	return format
+}
+
+func insertManyDailyTaskStats(many int, projectFmt string, requesterFmt string, taskNameFmt string, variantFmt string, distroFmt string, date time.Time, numSuccess, numFailed, numTimeout, numTestFailed, numSystemFailed, numSetupFailed int, avgDuration float64) error {
+
+	items := []interface{}{}
```
Comment (tychoish) [inline model/reliability/query_test.go]: this is a dumb microptimization, but I'd construct the slice as `make([]interface, 0, many)` which means that the underlying array is allocated with enough space for all of the items, and avoids reallocation

**Pull request merged**